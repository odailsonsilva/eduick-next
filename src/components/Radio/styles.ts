import styled, { css } from 'styled-components'

type WrapperItemProps = {
  isChecked: boolean
}

export const WrapperItem = styled.button<WrapperItemProps>`
  ${({ theme, isChecked }) => css`
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 5.6rem;
    align-items: center;
    justify-content: center;

    background: rgba(121, 85, 232, 0.3);
    padding: 1.4rem 1.6rem;

    width: fit-content;

    border-radius: ${theme.border.radius};

    border: none;

    ${isChecked &&
    css`
      border: 1px solid ${theme.colors.purple[300]};
      background: ${theme.colors.purple[700]};
    `}

    label {
      width: 100%;
      height: 5.6rem;
      font-size: 12px;
      font-weight: normal;
      display: flex;
      align-items: center;
      justify-content: center;

      input {
        display: none;
      }
    }
  `}
`

export const Check = styled.div`
  ${({ theme }) => css`
    width: 18px;
    height: 18px;
    border: none;
    border-radius: 50%;
    background: ${theme.colors.purple[300]};
    box-sizing: border-box;

    margin-right: 1.8rem;
    margin-bottom: 1px;

    border-radius: 50%;

    &::before {
      content: '';
      width: 100%;
      height: 100%;

      border-radius: 50%;
      background: ${theme.colors.yellow[500]} url('/icons/circle_checked.svg')
        center no-repeat;

      display: block;
      transform: scale(0);
      transition: all 0.1s;
    }
  `}
`

export const Label = styled.div`
  ${({ theme }) => css`
    p {
      font-size: 1.4rem;
      text-transform: uppercase;

      color: ${theme.colors.white[500]};
      font-weight: ${theme.font.bold};

      padding: 0;
    }
  `}
`

export const ContainerChecked = styled.div`
  display: flex;
  align-items: center;
`

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    place-content: center;

    width: fit-content;

    input:checked + ${ContainerChecked} {
      ${Check} {
        border-radius: 50%;
      }

      ${Check}::before {
        transform: scale(1);
      }
    }

    ${WrapperItem} + ${WrapperItem} {
      margin-left: ${theme.spacings.md};
    }
  `}
`
