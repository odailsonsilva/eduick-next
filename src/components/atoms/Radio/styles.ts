import styled, { css, DefaultTheme } from 'styled-components'
import media from 'styled-media-query'

type WrapperItemProps = {
  isChecked: boolean
}

export const WrapperItem = styled.button<WrapperItemProps>`
  ${({ theme, isChecked }) => css`
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 5.6rem;

    background: rgba(121, 85, 232, 0.3);

    width: fit-content;

    border-radius: ${theme.border.radius};

    border: none;

    ${isChecked &&
    css`
      border: 1px solid ${theme.colors.purple[300]};
      background: ${theme.colors.purple[700]};
    `}

    label {
      width: 100%;
      height: 5.6rem;
      font-size: 12px;
      font-weight: normal;
      display: flex;
      align-items: center;
      justify-content: center;
      padding: 1.4rem 1.6rem;
      cursor: pointer;

      input {
        display: none;
      }
    }

    ${media.lessThan('medium')`
      width: 100%;
    `}
  `}
`

export const Check = styled.div`
  ${({ theme }) => css`
    width: 18px;
    height: 18px;
    border: none;
    border-radius: 50%;
    background: ${theme.colors.purple[300]};
    box-sizing: border-box;

    margin-right: 0.8rem;
    margin-bottom: 1px;

    border-radius: 50%;

    &::before {
      content: '';
      width: 100%;
      height: 100%;

      border-radius: 50%;
      background: ${theme.colors.yellow[500]} url('/icons/circle_checked.svg')
        center no-repeat;

      display: block;
      transform: scale(0);
      transition: all 0.1s;
    }
  `}
`

export const Label = styled.div`
  ${({ theme }) => css`
    p {
      font-size: 1.4rem;
      text-transform: uppercase;

      color: ${theme.colors.white[500]};
      font-weight: ${theme.font.bold};

      padding: 0;

      ${media.lessThan('medium')`
        font-size: 1rem;
      `}
    }
  `}
`

export const ContainerChecked = styled.div`
  display: flex;
  align-items: center;
`

type WrapperProps = {
  direction: 'row' | 'column'
}

const modifiersWrapper = {
  row: (theme: DefaultTheme) => css`
    flex-direction: row;

    ${WrapperItem} + ${WrapperItem} {
      margin-left: ${theme.spacings.md};
    }
  `,
  column: (theme: DefaultTheme) => css`
    flex-direction: column;

    ${WrapperItem} + ${WrapperItem} {
      margin-top: ${theme.spacings.md};
    }
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, direction }) => css`
    display: flex;
    align-items: center;
    place-content: center;

    width: fit-content;

    input:checked + ${ContainerChecked} {
      ${Check} {
        border-radius: 50%;
      }

      ${Check}::before {
        transform: scale(1);
      }
    }

    ${media.lessThan('medium')`
      width: 100%;

      display: flex;

      ${modifiersWrapper[direction!](theme)}
    `}
  `}
`
