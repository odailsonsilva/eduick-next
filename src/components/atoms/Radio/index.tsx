import React from 'react'
import * as S from './styles'

type Options = {
  value: string
  label: string | any
}

type Props = {
  options: Options[]
  defaultValue?: string
}

const Radio = ({ options, defaultValue = '', ...rest }: Props) => {
  const [check, setCheck] = React.useState(defaultValue)

  return (
    <S.Wrapper>
      {options?.map((option) => (
        <S.WrapperItem
          type="button"
          key={option.value}
          isChecked={check === option.value}
        >
          <label htmlFor={option.value}>
            <input
              defaultChecked={option.value === defaultValue}
              value={option.value}
              type="radio"
              id={option.value}
              checked={check === option.value}
              onChange={(e) => setCheck(e.target.value)}
              {...rest}
            />
            <S.ContainerChecked>
              <S.Check />
              <S.Label>
                <p>{option.label}</p>
              </S.Label>
            </S.ContainerChecked>
          </label>
        </S.WrapperItem>
      ))}
    </S.Wrapper>
  )
}

export default Radio
