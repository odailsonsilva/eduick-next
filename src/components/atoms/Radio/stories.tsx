import { Story, Meta } from '@storybook/react/types-6-0'
import Radio from '.'

export default {
  title: 'Radio',
  component: Radio
} as Meta

const optionsRadio = [
  { value: '1', label: "I'M A TEACHER" },
  { value: '2', label: "I'M A STUDENT" }
]

export const Basic: Story = () => <Radio options={optionsRadio} />
Basic.parameters = {
  backgrounds: {
    default: 'dark'
  }
}
