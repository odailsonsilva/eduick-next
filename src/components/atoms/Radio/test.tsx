import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../../utils/tests/helpers'

import Radio from '.'

const optionsRadio = [{ value: '1', label: "I'M A TEACHER" }]

describe('<Radio />', () => {
  it('shold render default radio options', () => {
    const { container } = renderWithTheme(<Radio options={optionsRadio} />)
    const label = screen.getByText("I'M A TEACHER")

    expect(label).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
