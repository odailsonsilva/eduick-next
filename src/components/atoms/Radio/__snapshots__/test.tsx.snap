// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> shold render default radio options 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 5.6rem;
  background: rgba(121,85,232,0.3);
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 0.4rem;
  border: none;
}

.c2 label {
  width: 100%;
  height: 5.6rem;
  font-size: 12px;
  font-weight: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.4rem 1.6rem;
  cursor: pointer;
}

.c2 label input {
  display: none;
}

.c6 {
  width: 18px;
  height: 18px;
  border: none;
  border-radius: 50%;
  background: #A68EEF;
  box-sizing: border-box;
  margin-right: 0.8rem;
  margin-bottom: 1px;
  border-radius: 50%;
}

.c6::before {
  content: '';
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background: #FFD74F url('/icons/circle_checked.svg') center no-repeat;
  display: block;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
}

.c7 p {
  font-size: 1.4rem;
  text-transform: uppercase;
  color: #ffffff;
  font-weight: bold;
  padding: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  place-content: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .c1 + .c1 {
  margin-left: 2rem;
  margin-top: 0;
}

.c0 input:checked + .c3 .c5 {
  border-radius: 50%;
}

.c0 input:checked + .c3 .c5::before {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

@media (max-width:768px) {
  .c2 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c7 p {
    font-size: 1rem;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (max-width:350px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c0 .c1 + .c1 {
    margin-top: 2rem;
    margin-left: 0;
  }
}

<div
  class="c0"
>
  <button
    class="c1 c2"
    type="button"
  >
    <label
      for="1"
    >
      <input
        id="1"
        type="radio"
        value="1"
      />
      <div
        class="c3 c4"
      >
        <div
          class="c5 c6"
        />
        <div
          class="c7"
        >
          <p>
            I'M A TEACHER
          </p>
        </div>
      </div>
    </label>
  </button>
</div>
`;
